ocrmypdf: fails to be built/installed

### Describe the bug
```console
$ nix-env -iA nixos.ocrmypdf
[...]
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestcachePhase
pytestCheckPhase
Executing pytestCheckPhase
============================= test session starts ==============================
platform linux -- Python 3.8.9, pytest-6.2.3, py-1.10.0, pluggy-0.13.1
rootdir: /build/source, configfile: setup.cfg, testpaths: tests
plugins: helpers-namespace-2021.3.24, xdist-2.2.1, forked-1.3.0, cov-2.11.1
gw0 [316] / gw1 [316]m
.......F..xF........ss.................................................. [ 22%]
...............ss.................ss.ssss............................... [ 45%]
.............................................................s.......... [ 68%]
.......ss..........ssssssssssssssss..............................ss..... [ 91%]
............................                                             [100%]
=================================== FAILURES ===================================
____________________________ test_rasterize_rotated ____________________________
[gw0] linux -- Python 3.8.9 /nix/store/zv167zw7gwr9h0gpkfcikdfqnscrd0q6-python3-3.8.9/bin/python3.8

francais = (PosixPath('/build/source/tests/resources/francais.pdf'), <pikepdf.Pdf description='/build/source/tests/resources/francais.pdf'>)
outdir = PosixPath('/build/pytest-of-nixbld/pytest-0/popen-gw0/test_rasterize_rotated0')
caplog = <_pytest.logging.LogCaptureFixture object at 0x7fffefe0c940>

    def test_rasterize_rotated(francais, outdir, caplog):
        path, pdf = francais
        page_size_pts = (pdf.pages[0].MediaBox[2], pdf.pages[0].MediaBox[3])
        assert pdf.pages[0].MediaBox[0] == pdf.pages[0].MediaBox[1] == 0
        page_size = (page_size_pts[0] / Decimal(72), page_size_pts[1] / Decimal(72))
        target_size = Decimal('50.0'), Decimal('30.0')
        forced_dpi = Resolution(42.0, 4242.0)

        caplog.set_level(logging.DEBUG)
        rasterize_pdf(
            path,
            outdir / 'out.png',
            raster_device='pngmono',
            raster_dpi=Resolution(
                target_size[0] / page_size[0], target_size[1] / page_size[1]
            ),
            page_dpi=forced_dpi,
            rotation=90,
        )

        with Image.open(outdir / 'out.png') as im:
            assert im.size == (target_size[1], target_size[0])
>           assert im.info['dpi'] == (forced_dpi[1], forced_dpi[0])
E           assert (4242.0032, 42.0116) == (4242.0, 42.0)
E             At index 0 diff: 4242.0032 != 4242.0
E             Use -v to get the full diff

tests/test_ghostscript.py:77: AssertionError
------------------------------ Captured log call -------------------------------
DEBUG    ocrmypdf.subprocess:__init__.py:109 Running: ['gs', '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-sDEVICE=pngmono', '-dFirstPage=1', '-dLastPage=1', '-r16.666667x10.000000', '-o', '-', '-sstdout=%stderr', '-dAutoRotatePages=/None', '-f', '/build/source/tests/resources/francais.pdf']
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'IHDR' 16 13
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'iCCP' 41 2296
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:391 iCCP profile name b'default_gray.icc'
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:392 Compression method 0
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'pHYs' 2349 9
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'tEXt' 2370 31
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'IDAT' 2413 77
DEBUG    ocrmypdf._exec.ghostscript:ghostscript.py:126 Rotating output by 90
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'IHDR' 16 13
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'iCCP' 41 2291
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:391 iCCP profile name b'ICC Profile'
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:392 Compression method 0
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'pHYs' 2344 9
DEBUG    PIL.PngImagePlugin:PngImagePlugin.py:186 STREAM b'IDAT' 2365 106
_____________________________ test_rasterize_size ______________________________
[gw1] linux -- Python 3.8.9 /nix/store/zv167zw7gwr9h0gpkfcikdfqnscrd0q6-python3-3.8.9/bin/python3.8

francais = (PosixPath('/build/source/tests/resources/francais.pdf'), <pikepdf.Pdf description='/build/source/tests/resources/francais.pdf'>)
outdir = PosixPath('/build/pytest-of-nixbld/pytest-0/popen-gw1/test_rasterize_size0')

    def test_rasterize_size(francais, outdir):
        path, pdf = francais
        page_size_pts = (pdf.pages[0].MediaBox[2], pdf.pages[0].MediaBox[3])
        assert pdf.pages[0].MediaBox[0] == pdf.pages[0].MediaBox[1] == 0
        page_size = (page_size_pts[0] / Decimal(72), page_size_pts[1] / Decimal(72))
        target_size = Decimal('50.0'), Decimal('30.0')
        forced_dpi = Resolution(42.0, 4242.0)

        rasterize_pdf(
            path,
            outdir / 'out.png',
            raster_device='pngmono',
            raster_dpi=Resolution(
                target_size[0] / page_size[0], target_size[1] / page_size[1]
            ),
            page_dpi=forced_dpi,
        )

        with Image.open(outdir / 'out.png') as im:
            assert im.size == target_size
>           assert im.info['dpi'] == forced_dpi
E           assert (42.0116, 4242.0032) == Resolution(42.0x4242.0 dpi)
E             At index 0 diff: 42.0116 != 42.0
E             Use -v to get the full diff

tests/test_ghostscript.py:52: AssertionError
=========================== short test summary info ============================
FAILED tests/test_ghostscript.py::test_rasterize_rotated - assert (4242.0032,...
FAILED tests/test_ghostscript.py::test_rasterize_size - assert (42.0116, 4242...
======= 2 failed, 282 passed, 31 skipped, 1 xfailed in 266.66s (0:04:26) =======
builder for '/nix/store/j5msaqqpdavr44fnw73nz8bn9il83iqv-ocrmypdf-11.7.3.drv' failed with exit code 1
error: build of '/nix/store/j5msaqqpdavr44fnw73nz8bn9il83iqv-ocrmypdf-11.7.3.drv' failed
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. `nix-env -iA nixos.ocrmypdf`

### Expected behavior
Should be installed.

### Notify maintainers
@Kiwi 
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata
```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.52-hardened1, NixOS, 21.05.2001.d4590d21006 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.15`
 - channels(alexey): `""`
 - channels(root): `"nixos-21.05.2001.d4590d21006"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  ocrmypdf
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/tools/text/ocrmypdf/default.nix
+++ b/pkgs/tools/text/ocrmypdf/default.nix
@@ -30,14 +30,13 @@ let
 in
 buildPythonApplication rec {
   pname = "ocrmypdf";
-  version = "11.7.3";
-  disabled = ! python3Packages.isPy3k;
+  version = "12.3.0";
 
   src = fetchFromGitHub {
     owner = "jbarlow83";
     repo = "OCRmyPDF";
     rev = "v${version}";
-    sha256 = "0gs2w9kl5wwrs0hx2sivq3pdvpf3lkaifblwfbz5g31yl770blji";
+    sha256 = "122yv3p0v4fbx30zgppcznwnm7svg97gv0sa103xb6zcld68ggn2";
   };
 
   nativeBuildInputs = with python3Packages; [

